### PR TITLE
[Shortcuts]: Fix display keys

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/31625
Closes https://github.com/brave/brave-browser/issues/30879 (this is an alternative, much simpler solution)

The solution here is to not use the `key` property on the KeyboardEvent, and instead send the KeyCode to the backend, and convert it to a human friendly key there (we want the `DomKey` for the `DomCode`, not the key which would be typed).

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Add a new shortcut for any command with `Control`+`=`. `Control`+`=` should be displayed while entering the shortcut and after saving it.
2. Add a new shortcut for any command with `Control`+`Shift`+`=`. `Control`+`Shift`+`=` (not `+`) should be displayed while entering the shortcut and after saving it. 
3. On macOS, create a shortcut using `Alt`+`P`. `Alt`+`P` should be displayed in the new shortcut dialog and after saving.

### CODE REVIEW

**Review by sangwoo108 — APPROVED** [review]
LGTM. Can verify Alt + P appears as expected

image

File: browser/resources/settings/shortcuts_page/components/ConfigureShortcut.tsx [hunk 1]
```diff
@@ -119,10 +98,10 @@ export default function ConfigureShortcut(props: {
 
       // Enter on it's own is used to accept the current combination, if the
       // current keys are a valid shortcut.
-      if (e.code === "Enter" && !hasModifier && maxKeys.current.isValid()) {
+      if (e.code === "Enter" && !hasModifier && current.isValid) {
         props.onChange({
-          codes: keysToString(maxKeys.current.codes),
-          keys: keysToString(maxKeys.current.keys)
+          codes: keysToString(current.codes),
+          keys: keysToString([])
```
Comment (simonhong) [inline browser/resources/settings/shortcuts_page/components/ConfigureShortcut.tsx:104]: optional or nvm: maybe it should be like below as we send keys instead?
but still confusing which one is correct here `codes` or `keys` :) 
```suggestion
          keys: keysToString(current.codes)
```

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: browser/resources/settings/shortcuts_page/components/ConfigureShortcut.tsx (hunk 1 above)
Comment (fallaciousreasoning) [reply browser/resources/settings/shortcuts_page/components/ConfigureShortcut.tsx]: Good catch - the keys are displayed until we get a response from the browser process - I'd removed them while I was testing. Added back!

**Comment by kjozwiak** [comment]
Verification **`PASSED`** on `macOS 13.4.1 Ventura x64` using the following build(s):

```
Brave | 1.58.80 Chromium: 116.0.5845.51 (Official Build) nightly (x86_64)
--- | ---
Revision | 4408cc9795df260c5d5f033660ce757782b53b32
OS | macOS Version 13.4.1 (Build 22F82)
```

Using the STR/Cases outlined via https://github.com/brave/brave-core/pull/19431#issue-1821506053, ensured that adding the shortcuts mentioned worked without any issues/were also disabled correctly as per the following:

**`Example`** | **`Example`**
----------------|-----------------
Screenshot 2023-08-08 at 8 51 38 PM | Screenshot 2023-08-08 at 8 51 52 PM

**`Example`** | **`Example`**
----------------|-----------------
Screenshot 2023-08-08 at 8 52 32 PM | Screenshot 2023-08-08 at 8 52 43 PM

Screenshot 2023-08-08 at 3 40 31 PM

**Pull request merged**